### ISSUE TITLE
Fatal Error #8192 (E_DEPRECATED): preg_split(): Passing null to parameter #2 ($subject) of type string is deprecated

### ISSUE THREAD

**Issue by fiwswe** [mikepultz/netdns2#126]
In file: /var/www/pear/lib/Net/DNS2/Packet.php, line: 171

PHP 8.1.5, running the same code on PHP 8.0.18 works fine.
OpenBSD 7.1 stable amd64, PHP-FPM, Apache httpd (in case that makes any difference).
```
$ pear list   
Installed packages, channel pear.php.net:
=========================================
Package          Version State
Archive_Tar      1.4.9   stable
Console_Getopt   1.4.3   stable
Net_DNS2         1.5.0   stable
PEAR             1.10.12 stable
Structures_Graph 1.1.1   stable
XML_Util         1.4.5   stable
$ 
```

```
define('kTestDomain',
   'google.de');
define('kTestRRSet',
   'NS');
define('kTestNS',
   '194.50.187.134');

$resolver = new Net_DNS2_Resolver(array('nameservers'		=> array(kTestNS),
					'recurse'		=> false,
					'dnssec'		=> true,
					'strict_query_mode'	=> true,
					'local_host'		=> $_SERVER['SERVER_ADDR']));
$result = $resolver->query(kTestDomain,
			   kTestRRSet);

```
the $resolver->query(…) call fails!

**Comment by fiwswe** [comment]
Sorry, let me restate that a bit: The "Fatal" part was caused by my error handler that handled the deprecation warning. Still the issue should be fixed I think.

**Comment by mikepultz** [comment]
Fixed in v1.5.3
